src/toil/lib/conversions.py

```diff
@@ -24,14 +29,85 @@ def bytes_in_unit(unit: str = 'B') -> int:
         num_bytes = 1000 ** 3
     if unit.lower() in ['t', 'tb']:
         num_bytes = 1000 ** 4
+    if unit.lower() in ['p', 'pb']:
+        num_bytes = 1000 ** 5
+    if unit.lower() in ['e', 'eb']:
+        num_bytes = 1000 ** 6
     return num_bytes
 
 
 def convert_units(num: float,
                   src_unit: str,
-                  dst_unit: str) -> float:
+                  dst_unit: Optional[str] = 'B') -> float:
     """Returns a float representing the converted input in dst_units."""
-    units = ['B', 'KB', 'MB', 'GB', 'TB']
-    assert src_unit in units, f"{src_unit} not a valid unit, valid units are {units}."
-    assert dst_unit in units, f"{dst_unit} not a valid unit, valid units are {units}."
+    assert src_unit.lower() in VALID_UNITS, f"{src_unit} not a valid unit, valid units are {VALID_UNITS}."
+    assert dst_unit.lower() in VALID_UNITS, f"{dst_unit} not a valid unit, valid units are {VALID_UNITS}."
     return (num * bytes_in_unit(src_unit)) / bytes_in_unit(dst_unit)
+
+
+def human2bytes(string: str) -> int:
```

Comment:
I would split this into two functions.  One to parse and one to convert.  Something like:

```
def parse_memory_string(string: str) -> Tuple[float, str]:
    """Given a string representation of some memory (i.e. '1024 Mib'), return the number and unit."""
    for i, character in enumerate(string):
        # find the first character of the unit
        if character not in '0123456789.-_ ':
            units = string[i:].strip()
            assert units.lower() in VALID_UNITS, f"{units} not a valid unit, valid units are {VALID_UNITS}."
            return float(string[:i]), units
    return float(string), 'b'

def human2bytes(string: str) -> int:
    """Given a string representation of some memory (i.e. '1024 Mib'), return the integer number of bytes."""
    value, unit = parse_memory_string(string)
    return int(convert_units(value, src_unit=unit, dst_unit='b'))
```